SmartTutor/Controllers/ContentRepositoryService.cs

```diff
@@ -0,0 +1,129 @@
+﻿using SmartTutor.ContentModel;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SmartTutor.Controllers
+{
+    public class ContentRepositoryService
+    {
+        public ContentRepositoryService() { }
+
+        public EducationContent GetLongParameterListContent()
+        {
+            EducationContent educationContent = new EducationContent();
+
+            educationContent.EducationQuality = 4;
+            educationContent.EducationDifficulty = 4;
+            educationContent.EducationSnippets = new List<EducationSnippet>();
+
+            EducationSnippet snippetOne = new EducationSnippet();
+            EducationSnippet snippetTwo = new EducationSnippet();
+            EducationSnippet snippetThree = new EducationSnippet();
+            EducationSnippet snippetFour = new EducationSnippet();
+            EducationSnippet snippetFive = new EducationSnippet();
+
+            snippetOne.SnippetQuality = 4;
+            snippetOne.Tags = new List<Tag>(); snippetOne.Tags.Add(Tag.MustKnow);
+            snippetOne.EducationType = EducationType.ShortText;
+            snippetOne.SnippetDifficulty = 3;
+            snippetOne.Content = "This method have problem with long parameter list";
+            
+
+            snippetTwo.SnippetQuality = 5;
+            snippetTwo.Tags = new List<Tag>(); snippetTwo.Tags.Add(Tag.Interesting);
+            snippetTwo.EducationType = EducationType.LongText;
+            snippetTwo.Content = "The ideal number of arguments for a function is zero (niladic). Next comes one (monadic), followed closely by two (dyadic). Three arguments (triadic) should be avoided where possible. More than three (polyadic) requires very special justification—and then shouldn’t be used anyway.The argument is at a different level of abstraction than the function name and forces you to know a detail that isn’t particularly important.at that point.";
+            snippetTwo.SnippetDifficulty = 3;
+
+            snippetThree.SnippetQuality = 4;
```

Comment:
It would be useful if you encapsulate each snippet creation into a standalone private function. E.g.:

private EducationalSnippet CreateLongParameterListCodeSnippet() {
            EducationSnippet snippetFour = new EducationSnippet();
            snippetFour.SnippetQuality = 4;
            snippetFour.Tags = new List<Tag>(); snippetFour.Tags.Add(Tag.Interesting);
            snippetFour.EducationType = EducationType.CodeSnippet;
            snippetFour.Content = "Circle makeCircle(double x, double y, double radius);Circle makeCircle(Point center, double radius); ";
            snippetFour.SnippetDifficulty = 5;
            return snippetFour;
}
Just make sure not to call it snippetFour inside this private function :)

This will help us reuse the logic to create different content (e.g., with 2 or 3 snippets and with different ordering).

I would also use the Constructor to instantiate all these objects and place them in a Map, where the issueId is the key.

Reply:
I understand first part of this, but why we need these objects in a Map ?

Reply:
We can use other data structures, but a map (aka C# Dictionary) would give us quick access to educational content for a specific issue. We might also want the value part of the Dictionary to have a List of EduContent. The primary use case is to offer the first from the list (probably the highest quality one), but if the user should click "Give me next lesson" we would retrieve the 2nd element from the list, etc.